Выполнил свою часть задачи 1:

cryptojopa opened the pull request:
1) реализовал 2 сущности (Transaction и ExpenseCategory) и связь с БД с помощью ORM

cryptojopa committed:
Выполнил свою часть задачи 1:
1) реализовал 2 сущности (Transaction и ExpenseCategory) и связь с БД с помощью ORM

ivankrn commented on pom.xml:
```diff
@@ -47,6 +47,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
```
Не нужно специально добавлять jakarta в зависимости проекта. Она транзитивно подтягивается hibernate (до 11 java вроде бы даже поставлялась вместе с jdk).

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/ExpenseCategory.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.HashSet;
+import java.util.Set;
+
+@Entity
```
У нас категория транзакции, не категория трат. Следует переименовать и табличку, и саму сущность

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/ExpenseCategory.java:
```diff
@@ -0,0 +1,38 @@
+    private Set<Transaction> transaction = new HashSet<>();
+
+    @Column(nullable = false)
+    @Setter(AccessLevel.NONE)
```
Почему было принято решение не создавать сеттер для названия?

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/ExpenseCategory.java:
```diff
@@ -0,0 +1,38 @@
+    @Setter(AccessLevel.NONE)
+    private String categoryName;
+
+    /*
```
Закомментированные строки кода лучше убирать перед коммитом

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/Transaction.java:
```diff
@@ -0,0 +1,47 @@
+    @Setter(AccessLevel.NONE)
+    private Long id;
+
+    @Setter(AccessLevel.NONE)
```
Аналогично, почему не создаем сеттер для категории и всего остального?

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/Transaction.java:
```diff
@@ -0,0 +1,47 @@
+    @Setter(AccessLevel.NONE)
+    private BigDecimal amount = BigDecimal.ZERO;
+
+    @Column(nullable = false)
```
Добавить хоть какой-то геттер для валюты (см. пример из BankAccount.java)

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/Transaction.java:
```diff
@@ -0,0 +1,47 @@
+    private final String currency = "RUS";
+
+    /*
+    @ManyToOne
```
Аналогично, см. выше про закомментированные строки

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/Transaction.java:
```diff
@@ -0,0 +1,47 @@
+    @OneToOne
+    private BankAccount bankAccount;
+*/
+
```
Переопределить hashcode и equals

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/ExpenseCategory.java:
```diff
@@ -0,0 +1,38 @@
+    @OneToOne
+    private BankAccount bankAccount;
+*/
+
```
Переопределить hashcode и equals

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/FinanceTrackerApplication.java:
```diff
@@ -2,12 +2,13 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.GetMapping;
```
Убрать неиспользуемые импорты

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/ExpenseCategory.java:
```diff
@@ -0,0 +1,38 @@
+
+    @Column(nullable = false)
+    @Setter(AccessLevel.NONE)
+    private String categoryName;
```
Можно сократить просто до name (и так понятно, что это имя категории)

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/Transaction.java:
```diff
@@ -0,0 +1,47 @@
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "transaction_id_seq")
+    @Setter(AccessLevel.NONE)
+    private Long id;
+
```
Добавить поле с типом операции транзакции

ivankrn reviewed (changes requested):
Для первого раза очень хорошо! Вместе с тем нужно исправить небольшие замечания и доделать сущность транзакции

cryptojopa committed:
1 задание. Дополнение

cryptojopa committed:
1 задание. Поменял некорректные названия

cryptojopa committed:
Merge remote-tracking branch 'origin/task1' into task1-gleb

cryptojopa committed:
1 задание. Дополнение
1) Добавил связь между Transaction и BankAccount
2) Добавил поле с типом операции в Transaction.
3) Исправил аннотации у некоторых полей
4) Добавил геттер для валюты в  классе Transaction

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/BankAccount.java:
```diff
@@ -33,6 +35,9 @@ public class BankAccount {
     private final String currency = "RUS";
 
+    @OneToMany(mappedBy = "bankAccount")
+    private Set<Transaction> transaction = new HashSet<>();
```
Переименовать в transactions

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/OperationType.java:
```diff
@@ -0,0 +1,7 @@
+package ru.naumen.enterprisejavacourse.financetracker.database.model;
+
+public enum OperationType {
+
+    ADD,
+    SUBSTRACT
+}
```
Возможно стоило бы использовать существительные, т.к. у нас тип операции (т.е. заменить на addition и subtraction соответственно). Не критично, можно не исправлять

ivankrn commented on src/main/java/ru/naumen/enterprisejavacourse/financetracker/database/model/Transaction.java:
```diff
@@ -21,20 +22,30 @@ public class Transaction {
+    private OperationType operationType;
+
+    @ManyToOne
+    @JoinColumn(name = "bank_account_transaction")
```
1) Обычно внешние ключи оканчиваются на _id (https://vladmihalcea.com/manytoone-jpa-hibernate/, https://www.baeldung.com/hibernate-one-to-many#models). Не критично, но лучше исправить
2) Транзакция не может быть не привязана к какому-либо счету, лучше предусмотреть это и добавить либо optional=false в ManyToOne, либо в JoinColumn добавить nullable=false

cryptojopa committed:
1 задание. Дополнение
1) Переименовал Enums в OperationType
2) Переименовал список транзакций в BankAccount
3) Переименовал внешний ключ, добавил атрибут nullable со значением false

ivankrn committed:
Merge branch 'task1' into task1-gleb
